src/raftstore/store/worker/apply.rs

```diff
@@ -272,22 +272,20 @@ impl<'a> ApplyContextCore<'a> {
             enable_sync_log: false,
             sync_log_hint: false,
             exec_ctx: None,
-            use_delete_range: false,
         }
     }
 
-    pub fn enable_sync_log(mut self, eanbled: bool) -> ApplyContextCore<'a> {
-        self.enable_sync_log = eanbled;
-        self
+    pub fn push_exec_result(&mut self, res: ExecResult) {
+        self.exec_results.push(res);
     }
 
-    pub fn apply_res_capacity(mut self, cap: usize) -> ApplyContextCore<'a> {
-        self.apply_res = Vec::with_capacity(cap);
+    pub fn enable_sync_log(mut self, enabled: bool) -> ApplyContext {
+        self.enable_sync_log = enabled;
         self
     }
 
-    pub fn use_delete_range(mut self, use_delete_range: bool) -> ApplyContextCore<'a> {
-        self.use_delete_range = use_delete_range;
+    pub fn apply_res_capacity(mut self, cap: usize) -> ApplyContext {
```

Comment:
If you call `ApplyContext::new(...).apply_res_capacity(5)` it means you will allocate an empty vector, then drop it, then allocate a vector of size 5.

Maybe you can optimize this by having a `new_with_capacity()` that only allocates the vector once?